lib/events/filesessions/fileasync.go

```diff
@@ -466,13 +570,25 @@ func (u *Uploader) releaseSemaphore() error {
 }
 
 func (u *Uploader) emitEvent(e events.UploadEvent) {
-	if u.cfg.EventsC == nil {
-		return
-	}
+	// this channel is used by scanner to slow down/speed up
 	select {
-	case u.cfg.EventsC <- e:
-		return
+	case u.eventsCh <- e:
 	default:
 		u.log.Warningf("Skip send event on a blocked channel.")
 	}
```

Comment:
Since `eventsCh` is always serviced, remove the `default` case here or add a comment explaining why it's OK to drop the even and not adjust the rate limiter

Reply:
good point